### PR TITLE
Unhardcode some AI squad scan radii and misc AI improvements

### PR DESCRIPTION
**Pull request by reaperrr**
Increased configurability, improved disabled-trait-awareness and minor (though likely negligible) performance optimization.

Note: Intentionally didn't touch AirStates here, I'll wait for #14387 to be merged first.

### CODE REVIEW

File: OpenRA.Mods.Common/AI/States/StateBase.cs [hunk 1]
```diff
@@ -83,11 +86,17 @@ protected virtual bool ShouldFlee(Squad squad, Func<IEnumerable<Actor>, bool> fl
...
-			if (ownBaseBuildingAround.Any())
-				return false;
+			var randomSquadUnit = squad.Units.Random(squad.Random);
+			var dangerRadius = squad.Bot.Info.DangerScanRadius;
+			var units = squad.World.FindActorsInCircle(randomSquadUnit.CenterPosition, WDist.FromCells(dangerRadius)).ToList();
+
+			// If there are any own buildings within the DangerRadius, don't flee
+			// PERF: Avoid LINQ
+			foreach (var u in units)
+			{
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/AI/States/StateBase.cs]: Braces around single line statements.

**Comment by GraionDilach** [comment]
This looks like the solution of the issues I have with the AI since the isometric-range-rescale PR, so yes, pretty please fix the code issue.

**Comment by reaperrr** [comment]
Updated.

**Review by GraionDilach — APPROVED** [review]
:+1: then.

File: OpenRA.Mods.Common/AI/States/AirStates.cs [hunk 2]
```diff
@@ -63,13 +63,14 @@ protected static Actor FindDefenselessTarget(Squad owner)
...
 			var map = owner.World.Map;
+			var dangerRadius = owner.Bot.Info.DangerScanRadius;
 			detectedEnemyTarget = null;
-			var x = (map.MapSize.X % DangerRadius) == 0 ? map.MapSize.X : map.MapSize.X + DangerRadius;
-			var y = (map.MapSize.Y % DangerRadius) == 0 ? map.MapSize.Y : map.MapSize.Y + DangerRadius;
+			var x = (map.MapSize.X % dangerRadius) == 0 ? map.MapSize.X : map.MapSize.X + dangerRadius;
+			var y = (map.MapSize.Y % dangerRadius) == 0 ? map.MapSize.Y : map.MapSize.Y + dangerRadius;
 
-			for (var i = 0; i < x; i += DangerRadius * 2)
+			for (var i = 0; i < x; i += dangerRadius * 2)
```
Comment (pchote) [inline OpenRA.Mods.Common/AI/States/AirStates.cs]: Manually chunking the map like this is really awful :sad:

In the future we could win a lot of simplicity and perf by maintaining spatially partitioned maps of targetability and danger for each squad class.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/5738cc3048ec39c5e9cf1bb018b9843347c12088

**Pull request merged**